internal/mesh/internal/controllers/sidecarproxy/fetcher/data_fetcher.go

```diff
@@ -192,20 +107,24 @@ func (f *Fetcher) FetchExplicitDestinationsData(
 
 		// As Destinations resource contains a list of destinations,
 		// we need to find the one that references our service and port.
-		d.Explicit = findDestination(dest.ServiceRef, dest.Port, us.Destinations)
+		d.Explicit = findDestination(dest.ServiceRef, dest.Port, us.Data)
 
-		// Fetch ServiceEndpoints.
-		serviceID := resource.IDFromReference(dest.ServiceRef)
-		se, err := f.FetchServiceEndpoints(ctx, resource.ReplaceType(catalog.ServiceEndpointsType, serviceID))
+		var (
+			serviceID    = resource.IDFromReference(dest.ServiceRef)
+			serviceRef   = resource.ReferenceToString(dest.ServiceRef)
+			upstreamsRef = resource.IDToString(us.Resource.Id)
+		)
+
+		// Fetch Service
```

Comment:
This is the service with the same name as the one pointed to by the Upstreams item. It is the "logical" service you are trying to reach, **before** xRoute scrambling.

It no longer makes sense to examine `ServiceEndpoints` at  this layer, as that applies to the "real" services you reach on the other end of the xRoutes.